# Data Analysis with Python

## Introduction

This notebook demonstrates basic data analysis techniques using Python's pandas library. We will explore a dataset, perform statistical analysis, and create visualizations.

### Learning Objectives

- Load and explore datasets
- Perform descriptive statistics
- Create data visualizations
- Draw insights from data

## 1. Setting Up the Environment

First, we import the necessary libraries for our analysis.

In [13]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Configure display options
pd.set_option('display.max_columns', 10)
pd.set_option('display.width', 1000)

## 2. Creating Sample Data

For this demonstration, we'll create a sample dataset representing student performance.

In [14]:
# Create sample student data
data = {
    'Student ID': [1, 2, 3, 4, 5],
    'Math Score': [85, 78, 92, 70, 88],
    'Reading Score': [90, 82, 95, 75, 89],
    'Study Hours': [5, 3, 7, 2, 6]
}

df = pd.DataFrame(data)
df

,Student ID,Math Score,Reading Score,Study Hours
0,1,85,90,5
1,2,78,82,3
2,3,92,95,7
3,4,70,75,2
4,5,88,89,6


## 3. Descriptive Statistics

Let's examine the statistical summary of our dataset.

In [15]:
# Display descriptive statistics
df.describe()

,Student ID,Math Score,Reading Score,Study Hours
count,5.000000,5.00000,5.00000,5.000000
mean,3.000000,82.60000,86.20000,4.600000
std,1.581139,8.70632,7.79102,2.073644
min,1.000000,70.00000,75.00000,2.000000
25%,2.000000,78.00000,82.00000,3.000000
50%,3.000000,85.00000,89.00000,5.000000
75%,4.000000,88.00000,90.00000,6.000000
max,5.000000,92.00000,95.00000,7.000000


## 4. Data Analysis Functions

We can create reusable functions for common analysis tasks.

In [16]:
def calculate_correlation(df, col1, col2):
    """Calculate Pearson correlation between two columns."""
    return df[col1].corr(df[col2])

def get_performance_category(score):
    """Categorize performance based on score."""
    if score >= 90:
        return 'Excellent'
    elif score >= 80:
        return 'Good'
    elif score >= 70:
        return 'Average'
    else:
        return 'Needs Improvement'

In [17]:
# Analyze correlations
math_reading_corr = calculate_correlation(df, 'Math Score', 'Reading Score')
hours_math_corr = calculate_correlation(df, 'Study Hours', 'Math Score')

print(f"Correlation between Math and Reading scores: {math_reading_corr:.3f}")
print(f"Correlation between Study Hours and Math Score: {hours_math_corr:.3f}")

Correlation between Math and Reading scores: 0.986
Correlation between Study Hours and Math Score: 0.986


## 5. Key Findings

### Summary

Based on our analysis:

1. **Strong Positive Correlation**: Math and Reading scores are highly correlated (r = 0.985)
2. **Study Impact**: Study hours show strong positive correlation with performance
3. **Average Performance**: Mean math score is 82.6, reading score is 86.2

### Recommendations

- Encourage increased study time for improved performance
- Consider integrated math-reading curriculum
- Provide additional support for students scoring below 75

In [18]:
# Final summary
print("\n=== Analysis Complete ===")
print(f"Dataset: Student Performance")
print(f"Total Records: {len(df)}")
print(f"Analysis Date: 2026-03-18")


=== Analysis Complete ===
Dataset: Student Performance
Total Records: 5
Analysis Date: 2026-03-18
